Mathlib/Data/List/Infix.lean

```diff
@@ -210,6 +216,19 @@ theorem prefix_iff_eq_take : l₁ <+: l₂ ↔ l₁ = take (length l₁) l₂ :=
     fun e => e.symm ▸ take_prefix _ _⟩
 #align list.prefix_iff_eq_take List.prefix_iff_eq_take
 
+theorem prefix_take_iff {x y : List α} {n : ℕ}: x <+: y.take n ↔ x <+: y ∧ x.length ≤ n := by
+  constructor
+  { intro h; constructor
+    exact List.IsPrefix.trans h (List.take_prefix _ _)
+    replace h := h.length_le; simp[Nat.le_min] at h; exact h.left }
```

Comment:
You have a "non-terminal simp" here (the `simp[Nat.le_min] at h` is not closing a goal). These are harder to maintain over time: please replace them by an equivalent `simp only` call: writing `simp?` will tell you which lemmas you need. (This procedure is called "squeezing", for I think just historical reasons: just in case you hear that word and wonder.)